projects/Mallard/src/components/weather.tsx

```diff
@@ -0,0 +1,133 @@
+import { useCachedOrPromise } from 'src/hooks/use-cached-or-promise'
+import { withResponse } from 'src/helpers/response'
+import { Forecast } from '../common'
+import { metrics } from 'src/theme/spacing'
+import { WeatherIcon } from './weather/weatherIcon'
+import Moment from 'moment'
+import { fetchWeather } from 'src/helpers/fetch'
+
+const styles = StyleSheet.create({
+    weatherContainer: {
+        display: 'flex',
+        flexDirection: 'row',
+        width: 'auto',
+        height: 'auto',
+        paddingLeft: metrics.horizontal,
+        paddingRight: metrics.horizontal,
+    },
+    forecastItem: {
+        flex: 2,
+        height: 'auto',
+        borderLeftWidth: 1,
+        borderLeftColor: '#BDBDBD',
+        borderStyle: 'solid',
+        borderColor: '#BDBDBD',
+        justifyContent: 'center',
+        alignItems: 'flex-start',
+        paddingLeft: 14,
+        paddingTop: 2,
+    },
+    temperature: {
+        color: '#E05E00',
+        fontSize: 13,
+        fontFamily: 'DS3DisplaySans',
+    },
+    dateTime: {
+        color: '#000000',
+        fontSize: 13,
+        fontFamily: 'DS3DisplaySans',
+    },
+    locationNameContainer: {
+        flex: 5,
+        height: 60,
+        flexDirection: 'row',
+    },
+    locationName: {
+        fontFamily: 'TS3 Text Sans',
+        fontSize: 13,
+        color: '#000000',
+    },
+    locationPinIcon: {
+        fontFamily: 'guardianIcons_1.12',
+        fontSize: 30,
+        color: '#ACACAC',
+    },
+    root: {
+        fontFamily: 'weather',
+        alignItems: 'flex-start',
+        alignSelf: 'flex-start',
+        flexDirection: 'row',
+        justifyContent: 'flex-start',
+        flex: 0,
+        width: 'auto',
+    },
+})
+
+/**
+ * In future iterations we will use the device to access the user's lat/lon co-ordinates
+ * and make a request to http://mobile-weather.guardianapis.com/locations/v1/cities/geoposition/search?q=$LAT,$LON
+ * in order to fetch a more specific location code to make the below request. This will require permission from the user
+ * to use their location.
+ */
+
+const useWeatherResponse = () => {
+    const londonLocationCode = 328328
+    return withResponse<Forecast[]>(
+        useCachedOrPromise(
+            fetchWeather<Forecast[]>(
+                `forecasts/v1/hourly/12hour/${londonLocationCode}.json?metric=true&language=en-gb`,
+            ),
+        ),
+    )
+}
+
+const Weather = () => {
+    let weatherResponse = useWeatherResponse()
+
+    const renderWeather = (forecasts: Forecast[]) => {
+        if (forecasts && forecasts.length >= 9) {
+            return (
+                <View style={styles.weatherContainer}>
+                    <View style={styles.locationNameContainer}>
+                        <Text style={styles.locationPinIcon}>{'\uE01B'}</Text>
+                        <Text style={styles.locationName}>{'London'}</Text>
+                    </View>
+                    {[0, 2, 4, 6, 8].map(idx => {
```

Comment:
can you explain this in a comment?

Reply:
We ask for an hourly forecast for 12 hours from the API. The design asks for 5 forecasts every two hours. These are the array indexes for the right ones.

Reply:
Clarified with comment in code.